tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -1059,3 +1060,146 @@ def register(
+        fixed_image (:class:`numpy.ndarray`):
+            A fixed image.
+        moving_image (:class:`numpy.ndarray`):
+            A moving image.
+        fixed_mask (:class:`numpy.ndarray`):
+            A binary tissue mask for the fixed image.
+        moving_mask (:class:`numpy.ndarray`):
+            A binary tissue mask for the moving image.
+        grid_space:
+            Grid_space (mm) to decide control points.
+        scale_factors:
+            Scaling factor of each BSpline per level in a multi-level setting.
+        shrink_factor:
+            Shrink factor per level to change the size and complexity of the image.
+        smooth_sigmas:
+            Standard deviation for gaussian smoothing per level.
+        num_iterations:
+            Number of iterations.
+        sampling_percent:
+            Fraction of image used for metric evaluation.
+
+    Returns:
+        2D deformation transformation represented by a grid of control points.
+
+    """
+    fixed_image, moving_image = np.squeeze(fixed_image), np.squeeze(moving_image)
+    if len(fixed_image.shape) == 3 or len(moving_image.shape) == 3:
+        raise ValueError("The input images should be grayscale images.")
+
+    # Inverting intensity values
+    fixed_image_inv = 255 - fixed_image
+    moving_image_inv = 255 - moving_image
+
+    if len(fixed_mask.shape) > 2:
+        fixed_mask = fixed_mask[:, :, 0]
+    if len(moving_mask.shape) > 2:
+        moving_mask = moving_mask[:, :, 0]
+    fixed_mask = np.array(fixed_mask != 0, dtype=np.uint8)
+    moving_mask = np.array(moving_mask != 0, dtype=np.uint8)
+
+    # Background Removal
+    fixed_image_inv = cv2.bitwise_and(fixed_image_inv, fixed_image_inv, mask=fixed_mask)
+    moving_image_inv = cv2.bitwise_and(
+        moving_image_inv, moving_image_inv, mask=moving_mask
+    )
+
+    # Getting SimpleITK Images from numpy arrays
+    fixed_image_inv_sitk = sitk.GetImageFromArray(fixed_image_inv)
+    moving_image_inv_sitk = sitk.GetImageFromArray(moving_image_inv)
+    fixed_image_inv_sitk = sitk.Cast(fixed_image_inv_sitk, sitk.sitkFloat32)
+    moving_image_inv_sitk = sitk.Cast(moving_image_inv_sitk, sitk.sitkFloat32)
+
+    # Determine the number of BSpline control points using physical spacing
+    grid_physical_spacing = len(fixed_image.shape) * [
+        grid_space
+    ]  # A control point every grid_space (mm)
+    image_physical_size = [
+        size * spacing
+        for size, spacing in zip(
+            fixed_image_inv_sitk.GetSize(), fixed_image_inv_sitk.GetSpacing()
+        )
+    ]
+    mesh_size = [
+        int(image_size / grid_spacing + 0.5)
+        for image_size, grid_spacing in zip(image_physical_size, grid_physical_spacing)
+    ]
+    mesh_size = [int(sz / 4 + 0.5) for sz in mesh_size]
+
+    tx = sitk.BSplineTransformInitializer(
+        image1=fixed_image_inv_sitk, transformDomainMeshSize=mesh_size
+    )
+    print(f"Initial Number of Parameters: {tx.GetNumberOfParameters()}")
```

Comment:
At some point we need to go through the toolbox and replace prints etc. with proper logging.

Reply:
yes we need to do this especially for HPC jobs. Currently, outputs are saved in error files as mentioned in #255

Reply:
Replaced print with logging.info:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/1760d595678752b932649eaebbf677d9814bdd5a/tiatoolbox/tools/registration/wsi_registration.py#L1164